### DISCUSSION TITLE
[Chapter 04] Spring Boot 기본 WAS

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#25] [Chapter 04. Spring Web MVC Overview] [answered]
Spring Boot 프레임워크에는 Tomcat, Jetty, Undertow 등 여러 임베디드 WAS 종류가 있다. 

각각의 어떤 특징이 있을까? 

그리고 Spring WebFlux 는 Netty 를 기본 WAS 로 사용하는 이유가 궁금하다!

**Comment by corock** [comment, marked as answer]
Tomcat 과 Netty 의 가장 큰 차이점은 스레드를 단일로 사용하느냐 멀티로 운용하느냐인 것 같아요.
게다가 WebFlux 는 비동기 + _Non-blocking_ 콜라보를 통해 적은 리소스로 많은 트래픽을 처리할 수 있도록 설계되었는데,
둘의 콜라보가 좋아? Default 로 설정한 게 아닌가 싶습니다.
